src/datatrove/pipeline/extractors/base.py

```diff
@@ -114,13 +127,28 @@ def process_document(self, text, extract_fn):
         self._ensure_process(extract_fn)
         try:
             self.parent_conn.send(text)
-            if self.parent_conn.poll(timeout=self.timeout):
-                result = self.parent_conn.recv()
-                if isinstance(result, Exception):
-                    raise result
-                return result
-            else:
-                raise TimeoutError("Document extraction timed out")
+
+            deadline = time.time() + self.timeout
+            # loop with short sleeps instead of one big poll()
+            while True:
```

Comment:
do we actually need this to be in a loop? does it not work otherwise? because if the reason it works is having a high enough polling interval this won't work for the shorter intervals I normally use (1-2s)

Reply:
Yes we do, because if you timeout is 1 minutes. You want to check every 5 seconds for subprocess disconnect. This works with your short timeout just fine because: `poll_timeout = max(0, min(5, deadline - time.monotonic() + 0.1))` This will essentially set it to timeout if timeout < 5. Then you do just one cycle :)